cmp/pipelines/functional/eeg.py

```diff
@@ -256,215 +519,123 @@ def create_pipeline_flow(self, cmp_deriv_subject_directory, nipype_deriv_subject
-        
-        print('Parcellation atlas: ', parcellation_atlas)
-        print('Parcellation resolution: ', parcellation_resolution)
-        
-        if parcellation_atlas == 'lausanne2018':
-            parcellation_atlas = 'L2018'
-
-        if self.stages["EEGPreparer"].config.eeg_format == ".set":
-            # File with events/behavioral info
-            expe_name = datasource.inputs.EEG_params["expe_name"]
-            datasource.inputs.behav_file = [
-                os.path.join(
-                    self.base_directory,
-                    self.subject,
-                    "eeg",
-                    self.subject + "_task-" + expe_name + "_events.tsv",
-                )
-            ]
-
-            # Name of output file (EEG data source level)
-            datasource.inputs.epochs_fif_fname = os.path.join(
-                cmp_path_prefix_file,  f"{self.subject}_epo.fif"
-            )
-
-        # Name of output file (ROI time courses)
-        if parcellation_resolution is not None and parcellation_resolution != "":
-            datasource.inputs.roi_ts_file = os.path.join(
-                cmp_path_prefix_file,
-                f"{self.subject}_atlas-{parcellation_atlas}_res-{parcellation_resolution}_desc-epo_timeseries.npy"
-            )
-        else:
-            datasource.inputs.roi_ts_file = os.path.join(
-                cmp_path_prefix_file,
-                f"{self.subject}_atlas-{parcellation_atlas}_desc-epo_timeseries.npy"
-            )
 
-        datasource.inputs.output_query = dict()
+        # Create stages
+        preproc_flow = self.create_stage_flow("EEGPreprocessing")
+        esi_flow = self.create_stage_flow("EEGSourceImaging")
+        cmat_flow = self.create_stage_flow("EEGConnectome")
 
         # Data sinker for output
-        sinker = pe.Node(nio.DataSink(), name="eeg_datasinker")
-        sinker.inputs.base_directory = os.path.abspath(cmp_deriv_subject_directory)
+        sinker = self.create_datasinker_node(
+            output_directory=os.path.abspath(cmp_deriv_subject_directory)
+        )
 
         # Clear previous outputs
         self.clear_stages_outputs()
 
-        if "Cartool" in self.stages['EEGPreparer'].config.invsol_format:
-            # atlas image is required
-            parcellation = f'{self.subject}_atlas-{parcellation_atlas}_res-{parcellation_resolution}_dseg.nii.gz'
-            datasource.inputs.parcellation = os.path.join(
-                self.base_directory, 'derivatives', f'cmp-{__version__}',
-                self.subject, 'anat', parcellation
-            )
+        # BIDSDataGrabber node "datasource" is returning a
+        # list of file but the interfaces take only a file as input.
+        # This function allow to extract only one file to connect
+        # as input in the rest of the workflow.
+        def extract_first_file_from_list(files):
+            if len(files) > 1:
```

Comment:
The code in this condition is sorting a list of files if it is a list of multiple files (length>1) and the function returns the first element of the sorted list. Otherwise it returns the first and only element (length=0).